tools/diff-processor/diff/sets.go

```diff
@@ -0,0 +1,55 @@
+package diff
+
+import (
+	"sort"
+	"strings"
+)
+
+// Return the union of two maps, overwriting any shared keys with the second map's values
+func union[K comparable, V any](map1, map2 map[K]V) map[K]V {
+	if len(map1) == 0 {
+		return map2
+	}
+	if len(map2) == 0 {
+		return map1
+	}
+	merged := make(map[K]V, len(map1)+len(map2))
+	for k, v := range map1 {
+		merged[k] = v
+	}
+	for k, v := range map2 {
+		merged[k] = v
+	}
+	return merged
+}
+
+func sliceToSetRemoveZeroPadding(slice []string) map[string]struct{} {
+	set := make(map[string]struct{})
+	for _, item := range slice {
+		set[removeZeroPadding(item)] = struct{}{}
+	}
+	return set
+}
+
+// field1.0.field2 -> field1.field2
+func removeZeroPadding(zeroPadded string) string {
+	var trimmed string
+	for _, part := range strings.Split(zeroPadded, ".") {
+		if part != "0" {
+			trimmed += part + "."
+		}
+	}
+	if trimmed == "" {
+		return ""
+	}
+	return trimmed[:len(trimmed)-1]
+}
+
+func setToSortedSlice(set map[string]struct{}) []string {
```

Comment:
nit: I'm not sure it makes sense to split this out into a separate function since it's only used once. Not a blocker though.